deepwell/src/services/page/service.rs

```diff
@@ -565,6 +565,28 @@ impl PageService {
         Ok(page)
     }
 
+    /// Gets all deleted pages that match the provided slug.
+    pub async fn get_deleted_slug(
+        ctx: &ServiceContext<'_>,
+        site_id: i64,
+        slug: String,
```

Comment:
No reason for this to require an owned buffer.
```suggestion
        slug: &str,
```